code/game/objects/structures/low_wall.dm

```diff
@@ -0,0 +1,156 @@
+	update_overlays()
+
+/obj/structure/wall_frame/Destroy()
+	if(const_material.products_need_process())
+		STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/structure/wall_frame/process(delta_time)
+	if(!radiate())
+		STOP_PROCESSING(SSobj, src)
+		return
+
+/obj/structure/wall_frame/proc/radiate()
+	var/total_radiation = const_material.radioactivity
+	if(!total_radiation)
+		return
+
+	radiation_pulse(src, total_radiation)
+	return total_radiation
+
+/obj/structure/wall_frame/proc/set_material(var/new_material)
+	const_material = get_material_by_name(new_material)
+	if(!const_material)
+		qdel(src)
+		return
+	name = "[const_material.display_name] [initial(name)]"
+	max_health = round(const_material.integrity) //Should be 150 with default integrity (steel). Weaker than ye-olden Girders now.
+	health = max_health
+	if(applies_material_colour)
+		color = const_material.icon_colour
+	if(const_material.products_need_process()) //Am I radioactive or some other? Process me!
+		START_PROCESSING(SSobj, src)
+	else if(datum_flags & DF_ISPROCESSING) //If I happened to be radioactive or s.o. previously, and am not now, stop processing.
+		STOP_PROCESSING(SSobj, src)
+
+/obj/structure/wall_frame/update_overlays()
+	overlays.len = 0
+
+	color = const_material.icon_colour
+
+	var/image/smoothed_stripe = image(const_material.wall_stripe_icon, icon_state, layer = ABOVE_WINDOW_LAYER)
+	smoothed_stripe.appearance_flags = RESET_COLOR
+	smoothed_stripe.color = stripe_color || const_material.icon_colour
+	overlays += smoothed_stripe
+
+	if(!GLOB.wallframe_typecache)
+		GLOB.wallframe_typecache = typecacheof(list(/obj/machinery/door/airlock))
+	var/neighbor_stripe = NONE
+	for(var/cardinal in GLOB.cardinal)
+		var/turf/step_turf = get_step(src, cardinal)
+		var/obj/structure/wall_frame/neighbor = locate() in step_turf
+		if(neighbor)
+			continue
+		if(!can_area_smooth(step_turf))
+			continue
+		for(var/atom/movable/movable_thing as anything in step_turf)
+			if(GLOB.wallframe_typecache[movable_thing.type])
+				neighbor_stripe ^= cardinal
+				break
+		if(GLOB.wallframe_typecache[step_turf.type])
+			neighbor_stripe ^= cardinal
+
+	if(neighbor_stripe)
+		var/image/neighb_stripe_overlay = new ('icons/turf/walls/neighbor_stripe.dmi', "stripe-[neighbor_stripe]", layer = ABOVE_WINDOW_LAYER)
+		neighb_stripe_overlay.appearance_flags = RESET_COLOR
+		neighb_stripe_overlay.color = stripe_color || const_material.icon_colour
+		overlays += neighb_stripe_overlay
+		if(shiny_stripe)
+			var/image/shine = image('icons/turf/walls/neighbor_stripe.dmi', "shine-[smoothing_junction]")
+			shine.appearance_flags = RESET_COLOR
+			overlays += shine
+
+	return ..()
+
+/obj/structure/wall_frame/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(air_group || (height==0)) return 1
+	if(istype(mover,/obj/projectile))
+		return 1
+	if(istype(mover) && mover.check_pass_flags(ATOM_PASS_TABLE))
+		return 1
+
+/obj/structure/wall_frame/attackby(var/obj/item/W, var/mob/user)
```

Comment:
please make sure /attackby params are the same as /atom/attackby's params, we want them to be coherent.